add some GitHub workflows for flake8 and add black dependecy

bufferoverflow opened the pull request:
I'm thinking about deploying litellm and started to review the code a bit. Then I figured out that flake8 is ignoring a bunch of rules and while reviewing that I saw that black will solve a bunch of those already. Let me know what you think about the direction. Of course, I can create some individual PRs for those changes. And I'm ready to contribute some follow-up PRs to reduce the .flake8 ignore list.

bufferoverflow committed:
feat: add lint workflow

bufferoverflow committed:
chore: add flake8 as dev dependency

bufferoverflow committed:
chore: add black as dev dependency

bufferoverflow committed:
chore: list all ignored flake8 rules explicit

bufferoverflow committed:
chore: add pytest as dev dependency

bufferoverflow committed:
docs: add flake8 to the contributing section

bufferoverflow commented:
@krrishdholakia Could you have a look at this suggestion, I'm ready to rework towards your preference and create dedicated PRs

krrishdholakia commented:
Hi @bufferoverflow this is awesome. I'm happy to move in this direction. 

Your PR makes 160+ file changes. A quick look show them to mainly be linting changes, but it's hard to tell. 

2 asks: 
1. Can we isolate the diff to just be the change you intend to make 
2. What linter are you using? I'm currently using pylint in vscode, but happy to try another one if there's a preferred alternative.

bufferoverflow committed:
chore: list all ignored flake8 rules explicit

bufferoverflow commented:
My understanding was that your are using flake8 as there was a `.flake8` file so I started from there by adding it to `pyproject.toml` and excluding rules explicit(I will contribute more to remove a bunch of them as soon as this is in).

pylint and flake8 are quite similar, just run `poetry run flake8`, I added this also to the `README.md` and the github workflow.

Regarding formatting, [black](https://black.readthedocs.io/en/stable/) is an opinionated formatter we e.g., are using on many projects. The cool thing, no discussion about formatting anymore. Within CI we could run `poetry run black --check`  and to format the source you just run `poetry run black .` done. I added the list of flake8 excludes which can be removed at a later step when using black, just added the dev dependency so you can get familar with black.
